core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -99,10 +99,25 @@ public void run() {
                   store.setStatus(tid, IN_PROGRESS);
                 }
                 op = executeCall(tid, op);
+                // It's possible that a Fate operation impl
+                // may not do the right thing with an
+                // InterruptedException.
+                if (Thread.currentThread().isInterrupted()) {
+                  throw new InterruptedException("Fate Transaction Runner thread interrupted");
+                }
               } else {
                 continue;
               }
-
+            } catch (InterruptedException e) {
```

Comment:
Could move this check down into the `catch(Exception e){` block and there can see if the exception or its cause is an interrupted exception.   Like [this function](https://github.com/apache/accumulo/blob/960bd08820d2f1477ba7241a04dd576fbd78fb20/core/src/main/java/org/apache/accumulo/core/util/ShutdownUtil.java#L39) does does for IOException.  Sometimes code will wrap an Interrupted exception w/ some kind of runtime exception.

Also, I do not know if when this catch block rethrows the exception if it will go to the next catch block, do not think it will but not sure.  If not then it will not transition to failed.  That could be another reason to move the check into the `catch (Exception e)` block.  When it is in that block it can decide if tansition to failed should be called or not.